Kernel/BestLiteralSelector.hpp

```diff
@@ -261,14 +261,19 @@ class CompleteBestLiteralSelector
   {
     CALL("CompleteBestLiteralSelector::canBeSelected");
 
-    FirstOrderSubtermIt fsi(lit);
-
-    while(fsi.hasNext()){
-      TermList t = fsi.next();
-      if(t.isVar() && maxTermHeads->contains(t.var())){
-        return false;
-      }
-    }
+    // TODO: 
+    // - fsi always returns terms only.therefore the whole implementation would always return true anyways, so we can comment out this code.
+    // - this was discovered during a refactoring. Why was this function here in the first place? 
+    // it is definitely dead/unnecessary code at the time of refactoring
+    //
+    // FirstOrderSubtermIt fsi(lit);
+    //
+    // while(fsi.hasNext()){
+    //   TermList t = TermList(fsi.next());
+    //   if(t.isVar() && maxTermHeads->contains(t.var())){
+    //     return false;
+    //   }
+    // }
```

Comment:
@ibnyusuf Any thoughts on this one?

Reply:
Yes, this is a higher-order concern. If you look at the calls to the function, they are only made when `combSup` is `true`, so first-order proof search should not be affected. 

In the higher-order calculus, due to some wierd requirement of the completeness proof, we should not be selecting a literals ` s = t` if `s` has  a variable head ` x` and `x` also occurs unapplied in the literal (or clause, I dont remember exactly!)

In any case, if you feel confident that it shouldn't interact with first-order stuff, it would be good if the code could be reset to old version.

Reply:
Ahmed - maybe I misunderstood you, but I think Joe's saying this should be equivalent to `return true`, because the `FirstOrderSubtermIt` returns only non-variable terms. I think this is true even in the HOL case - should this use a different iterator or something instead?

Reply:
Which is of course 100% true. This means that the code above has never carried out the role taht I thought it did. Let's leave it commented out for now until I find the time to decide what to do here. Thanks for bringing this to light!

Reply:
okidoke then let's leave it as is for now :)